src/containers/pages/FocusInvestigation/map/active/tests/index.test.tsx

```diff
@@ -116,4 +125,45 @@ describe('containers/pages/FocusInvestigation/activeMap', () => {
     expect(wrapper.find('GisidaWrapperMock').props()).toMatchSnapshot();
     wrapper.unmount();
   });
+
+  it('calls superset with the correct params', async () => {
+    const mock: any = jest.fn();
+    const supersetMock: any = jest.fn();
+    supersetMock.mockImplementation(async () => []);
+    store.dispatch(fetchGoals([fixtures.goal3]));
+    store.dispatch(fetchJurisdictions([fixtures.jurisdictions[0]]));
+    store.dispatch(fetchPlans([fixtures.plan1 as Plan]));
+    store.dispatch(fetchTasks(fixtures.tasks));
+    const props = {
+      currentGoal: fixtures.goal3,
+      history,
+      location: mock,
+      match: {
+        isExact: true,
+        params: { id: fixtures.plan1.id },
+        path: `${FI_SINGLE_URL}/:id`,
+        url: `${FI_SINGLE_URL}/13`,
+      },
+      pointFeatureCollection: wrapFeatureCollection([fixtures.coloredTasks.task3.geojson]),
+      polygonFeatureCollection: wrapFeatureCollection([fixtures.coloredTasks.task2.geojson]),
+      supersetService: supersetMock,
+    };
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <ConnectedMapSingleFI {...props} />
+        </Router>
+      </Provider>
+    );
+    // HACK: HACK: to force for asynchronous code in component to execute.
+    await Promise.resolve();
+    await Promise.resolve();
+    await Promise.resolve();
+    await Promise.resolve();
+    await Promise.resolve();
+    expect(supersetMock).toHaveBeenCalledTimes(5);
+    const callList = [[1, {}], [2, {}], [0, {}], [3, {}], [4, {}]];
```

Comment:
Why are these objects empty?  I think in the code we call each with some params, no? e.g. `supersetService(SUPERSET_JURISDICTIONS_SLICE, jurisdictionsParams)`

Reply:
I mocked out the response of `superset.getFormData` to return an empty object. The thought process was that its functionality is already tested in the js-tools supersetConnector. So essentially I just needed to test `supersetFetch` gets called with the right args irregardless of what `superset.getFormData` response is. 

Calling the __supersetFetch mock__ with  an empty object means that:
 - `superset.getformData` was called, the test assumes the response of `getFormData` is already tested and therefore it mocks out the actual response with an empty object

I think what I could add is to check that a mock of `superset.getFormData` is called the right number of times and with the correct args i.e. e.g for a single call:

```typescript 
    [ 
        3000, [
               { comparator: plan.plan_id, operator: '==', subject: 'plan_id' },
           ]
     ]
``` 

but I still think that I would retain `superset.getFormData` mock implementation to return an empty dictionary, as it currently does.

@moshthepitt  How would you advise?

Reply:
> So essentially I just needed to test supersetFetch gets called with the right args irregardless of what superset.getFormData response is.

I agree with this line of thinking.

It seems to me, however, that our current tests do not achieve this goal.

This bit of code `supersetService(SUPERSET_JURISDICTIONS_SLICE, jurisdictionsParams)` results in supersetFetch being called with two params, the first one is the slice, the second one is the filter.

We are therefore not testing that we are indeed using the correct (or any) filters.  We need to ensure that the right value of the filter is captured in the test.  Do you agree?

I think that we need to find a way to mock `superset.getFormData` in the same test so that we have a more useful and accurate test.